### DISCUSSION TITLE
Path alias for server routes not resolving correctly with layer feature.

### DISCUSSION THREAD

**Discussion by azoom-y-ishii** [nuxt/nuxt#27298] [Questions]
I'm using the Layer feature, but while the '@' path alias resolves correctly in components, it doesn't work in Server Routes.

Here is a minimal reproduction:
https://stackblitz.com/edit/nuxt-starter-d6xolj

This project extends the my-layer folder as a layer. In `my-layer/components/HelloWorld.vue`, `import { message } from '@/constants/foo'` imports without any error, but in `server/api/ng.get.ts`, it causes an error. 

 ```
Error: Could not load /home/projects/nuxt-starter-d6xolj/constants/foo (imported by my-layer/server/api/ng.get.ts): ENOENT: no such file or directory, open '/home/projects/nuxt-starter-d6xolj/constants/foo'
```

To resolve this issue, I have created the following alias, which works correctly in `server/api/ok.get.ts`:

```
alias: {
  '@my-layer/constants': path.resolve(__dirname, 'constants'),
},
```
Is this error the expected behavior of Nuxt? It seems odd that it only occurs with Server Routes.

**Comment by angelhdzmultimedia** [comment]


**Reply by Zebnastien** [reply]
Hi,

I'm not sure if this is related, but I don't understand how to use relative path with 'fs' and layers.

I am trying to read a .sql files from a Nuxt server api.
I'm using Nuxt 4 (future), and I have an api in /layers/test/server/api/index.get.ts
The problem is that the working folder is the root of the project, not the root of the layer.
The consequence is that I have to use two different paths depending on whether the layer is used standalone or as a layer of an app.

I don't know if I'm being clear, but I can give more details if necessary.
